src/main/java/de/tum/cit/aet/artemis/atlas/dto/metrics/CompetencyExerciseMasteryCalculationDTO.java

```diff
@@ -7,6 +7,6 @@
 import de.tum.cit.aet.artemis.exercise.domain.DifficultyLevel;
 
 @JsonInclude(JsonInclude.Include.NON_EMPTY)
-public record CompetencyExerciseMasteryCalculationDTO(double maxPoints, DifficultyLevel difficulty, boolean isProgrammingExercise, Double lastScore, Double lastPoints,
-        Instant lastModifiedDate, long submissionCount) {
+public record CompetencyExerciseMasteryCalculationDTO(double maxPoints, DifficultyLevel difficulty, boolean isProgrammingExercise, double competencyLinkWeight, Double lastScore,
```

Comment:
Is it intended that `competencyLinkWeight` is changed to `double` instead of `Double`? Why is that the case?

Reply:
The weight is newly added and not changed. The lastScore can be null (student did not participate yet) but the link weight has to be set, otherwise the data is corrupted